examples/custom_signaller/src/direct_message.rs

```diff
@@ -0,0 +1,50 @@
+use iroh::{endpoint::Connection, protocol::ProtocolHandler, Endpoint, PublicKey};
+use matchbox_socket::PeerEvent;
+
+use crate::get_timestamp;
+
+pub const DIRECT_MESSAGE_ALPN: &[u8] = b"/matchbox-direct-message/0";
+#[derive(Debug, Clone)]
+pub struct DirectMessageProtocol(pub async_broadcast::Sender<(PublicKey, PeerEvent)>);
+
+type DirectMessage = (PeerEvent, u128);
+
+impl DirectMessageProtocol {
+    async fn handle_connection(self, connection: Connection) -> anyhow::Result<()> {
+        let _remote_node_id = connection.remote_node_id()?;
+        let mut recv = connection.accept_uni().await?;
+        let data = recv.read_to_end(1024 * 63).await?;
+        let _ = recv.stop(iroh::endpoint::VarInt::from(0_u8));
```

Comment:
Is there a particular reason why you call `stop` after `read_to_end`?

If `read_to_end` finishes, either the stream is already stopped (because it was read to end) or it errors out with a `TooLong` error, in which case the `.stop` line isn't executed at all.